### ISSUE TITLE
Extending Enums

### ISSUE THREAD

**Issue by mttrbrts** [accordproject/concerto#291]
# Feature Request 🛍️

I want to prompt a discussion about the extension of enumerated types as a feature in a type systems

## Use Case

Consider the following...
```
enum Foo {
 o A
 o B
}

enum Bar extends Foo {
  o C
}
```

## Alternative

An alternative solution would be to allow a users define new types through copying rather than extension. E.g.

```
enum Bar {
  ...Foo
  o C
}
```

## Context

This discussion for the same feature in TypeScript is useful reference, https://github.com/microsoft/TypeScript/issues/17592

In summary, the main objection seems to be the loss of the "closed" property of Enums. With enum extension, client code looses the quality that it can be sure of complete coverage of members.

**Comment by jeromesimeon** [comment]
Indeed we could use inheritance. The cost being some complexity in code using enums that may have to support annotations when building values. How important is this feature really?

**Comment by mttrbrts** [comment]
Closing due to lack of motivating use cases
